src/algorithm/bid_argmax_profit.jl

```diff
@@ -0,0 +1,102 @@
+"""
+    function bid_argmax_profit(args...; kwargs)
+
+Calculates the bid that returns the maximum profit of the slack bus based on an aproximation
+of the profit function which is calculated using the Transmission Constrained Residual
+Demand Derivative (TCRDD). A plot of this approximation can be displayed.
+
+Detailed description: The algorithm considers the Locational Marginal Price (LMP) for a
+determined bid in the slack generator which matches to a single point of the real profit
+function. Then, using the TCRDD of that specific point, an approximation of the real profit
+function is build. The function then evaluates the aproximated function from the minimum
+active power (Pmin) to the maximum active power (Pmax) of the slack generator. The bid that
+provides the maximum profit is returned.
+
+# Arguments
+- `sys::System`:                            Power system in p.u. (from PowerSystems.jl)
+- `res::PowerSimulations.
+    OperationsProblemResults`:              Results of the solved OPF for the system
+                                             (from PowerSimulations.jl)
+- `BaseMVA::Float64`:                       Apparent Power Base for the system [MVA]
+- `lmp::DataFrame`:                         LMP of all buses of the system
+- `tcrdd_slack::Float64`:                   TCRDD [pu]
+- `bid::Float64`:                           Bid for the slack generator [pu]
+- `Pmin_orig::Float64`:                     Active power minimum limit [pu]
+- `Pmax_orig::Float64`:                     Active power maximum limit [pu]
+
+# Keywords
+- `segm_bid_argmax_profit::Int64 = 50000:   Segments to evaluate the approx profit function
+- `print_plots::Bool = true`:               Flag to print plots
+
+# Throws
+- `NotFoundError`:                          No Slack bus found in the system
+"""
+function bid_argmax_profit(
```

Comment:
Could we have consistent casing for parameters? We're a bit all over the place between:

```
sys
BaseMVA
Pmin_orig
```

Would be nice to follow Julia conventions of:

```
sys
base_mva
pmin_orig
```